frontend/src/download.js

```diff
@@ -42,6 +42,9 @@ $(document).ready(function() {
   });
   $('#download-btn').click(download);
   function download() {
+    // Disable the download button to avoid accidental double clicks.
+    $('#download-btn').attr('disabled', 'disabled');
```

Comment:
Maybe we can store this selector (`#download-btn`) in a var because it seems we try to retrieve this ID a few times
so something like this would be great : 
```javascript
const $downloadBtn = $('#download-btn');
```